### ISSUE TITLE
Issue with Mach-O File in 'libuniffi_scanner.a' during iOS SDK Integration

### ISSUE THREAD

**Issue by pniteshnoble** [Inateck-Technology-Inc/ios_sdk#1]
I am facing an issue when attempting to integrate the iOS SDK (ios_sdk) into my project. The error message states: "Archive member '/' not a mach-o file in '/Users/user/Downloads/ios_sdk-master/ios_sdk/lib/libuniffi_scanner.a'". This error occurs during the build process and prevents successful compilation.

Steps to Reproduce:

1. Download the iOS SDK from the official repository.
2. Attempt to build the project.

Error:
The build process fails with the error message: "Archive member '/' not a mach-o file in '/Users/user/Downloads/ios_sdk-master/ios_sdk/lib/libuniffi_scanner.a'".

Additional Information:

Xcode Version: 15.1
iOS SDK Version: 17.2

Attached is a screenshot of the error encountered during the build process.
Screenshots:

<img width="1552" alt="Screenshot 2024-02-29 at 1 35 51 PM" src="https://github.com/Inateck-Technology-Inc/ios_sdk/assets/161555114/0c7a6bcd-8a9f-4512-aa96-77edb866106b">

**Comment by kantilalpatel** [comment]
@pniteshnoble Did you able to figure out the solution for this?

**Comment by pniteshnoble** [comment]
> @pniteshnoble Did you able to figure out the solution for this?

Not yet, I will surly post here, If I found any solution.

**Comment by soSeven** [comment]
The new version is available for use.

**Comment by bentayloruk** [comment]
@soSeven this is great! I was literally writing to ask if anyone had a solution to this and your comment popped up in front of me! I'll give it a go now.

**Comment by bentayloruk** [comment]
@soSeven I've got the latest version and am on the `master` branch, but I get a lot of build errors. Should the project build "out of the box" or are there steps I need to take before it will work? I'm not finding the README very clear on this point. I'm a newbie to XCode and iOS development, but have years of general programming experience, so sorry if this is a dumb question!

```
Undefined symbol: enum case for InateckScannerBleKit.BLEManager.ScanStatus.scan(InateckScannerBleKit.BLEManager.ScanStatus.Type) -> (InateckScannerBleKit.BLEDevice) -> InateckScannerBleKit.BLEManager.ScanStatus

Undefined symbol: enum case for InateckScannerBleKit.BLEManager.ScanStatus.stop(InateckScannerBleKit.BLEManager.ScanStatus.Type) -> InateckScannerBleKit.BLEManager.ScanStatus

Undefined symbol: enum case for InateckScannerBleKit.BLEManager.ScanStatus.start(InateckScannerBleKit.BLEManager.ScanStatus.Type) -> InateckScannerBleKit.BLEManager.ScanStatus

Undefined symbol: type metadata accessor for InateckScannerBleKit.BLEManager.ScanStatus

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEManager.scanDevices(timeoutAfter: Swift.Double, completion: (InateckScannerBleKit.BLEManager.ScanStatus) -> ()) -> ()

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEManager.disconnectHandler.setter : ((InateckScannerBleKit.BLEDevice) -> ())?

Undefined symbol: static InateckScannerBleKit.BLEManager.shared.getter : InateckScannerBleKit.BLEManager

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEManager.devices.getter : [InateckScannerBleKit.BLEDevice]

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEManager.stopScan() -> ()

Undefined symbol: type metadata accessor for InateckScannerBleKit.BLEManager

Undefined symbol: InateckScannerBleKit.MessageManager.getVersion((Swift.Result<Swift.String, Swift.Error>) -> ()) -> ()

Undefined symbol: InateckScannerBleKit.MessageManager.getBatteryInfo((Swift.Result<Swift.Int, Swift.Error>) -> ()) -> ()

Undefined symbol: InateckScannerBleKit.MessageManager.getSettingInfo((Swift.Result<(Swift.String, Foundation.Data), Swift.Error>) -> ()) -> ()

Undefined symbol: InateckScannerBleKit.MessageManager.setSettingInfo(with: Swift.String, completion: (Swift.Result<(Swift.String, Foundation.Data), Swift.Error>) -> ()) -> ()

Undefined symbol: InateckScannerBleKit.MessageManager.getHardwareInfo((Swift.Result<Swift.String, Swift.Error>) -> ()) -> ()

Undefined symbol: InateckScannerBleKit.BLEDevice.disconnect(completion: (Swift.Result<(), Swift.Error>) -> ()) -> ()

Undefined symbol: enum case for InateckScannerBleKit.BLEDevice.ConnectType.connecting(InateckScannerBleKit.BLEDevice.ConnectType.Type) -> InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: enum case for InateckScannerBleKit.BLEDevice.ConnectType.disconnected(InateckScannerBleKit.BLEDevice.ConnectType.Type) -> InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: enum case for InateckScannerBleKit.BLEDevice.ConnectType.disconnecting(InateckScannerBleKit.BLEDevice.ConnectType.Type) -> InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: enum case for InateckScannerBleKit.BLEDevice.ConnectType.none(InateckScannerBleKit.BLEDevice.ConnectType.Type) -> InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: enum case for InateckScannerBleKit.BLEDevice.ConnectType.connected(InateckScannerBleKit.BLEDevice.ConnectType.Type) -> InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: type metadata accessor for InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEDevice.connectState.getter : InateckScannerBleKit.BLEDevice.ConnectType

Undefined symbol: InateckScannerBleKit.BLEDevice.messageManager.getter : InateckScannerBleKit.MessageManager

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEDevice.name.getter : Swift.String?

Undefined symbol: dispatch thunk of InateckScannerBleKit.BLEDevice.uuid.getter : Swift.String

Undefined symbol: InateckScannerBleKit.BLEDevice.connect(timeout: Swift.Double, completion: (Swift.Result<(), Swift.Error>) -> ()) -> ()

Undefined symbol: type metadata accessor for InateckScannerBleKit.BLEDevice

Undefined symbol: nominal type descriptor for InateckScannerBleKit.BLEDevice

Linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by bentayloruk** [comment]
From the build log, it looks like this is due to a CPU architecture config issue or something. Posting here in case this helps, but will investigate (and learn!) further. Will raise as a new issue if I can't get it going.
<img width="840" alt="image" src="https://github.com/Inateck-Technology-Inc/ios_sdk/assets/189409/a27608a6-f843-4b66-b187-ee13c98a4c28">

**Comment by soSeven** [comment]
Please run the demo on a real device, not on an emulator.

**Comment by bentayloruk** [comment]
Thank you for the reply @soSeven! I figured this out a little while later and have the app running on my phone. Unfortunately, my BCST-73 requires a firmware update to work with the SDK, so I'm waiting on Inateck support to provide the firmware file. I was suprised it is not possible to download it.
